lib/providers/race_events.dart

```diff
@@ -39,3 +42,71 @@ Stream<Map<String, List<int>>> teamScores(TeamScoresRef ref, String eventId) {
   });
   return controller.stream;
 }
+
+// TODO change to immutable map
+class TeamPositionNotifier extends StateNotifier<Map<String, LatLng>> {
+  final DatabaseReference firebaseDb;
+
+  TeamPositionNotifier(this.firebaseDb, String eventId, List<String> teams) : super(<String, LatLng>{}) {
+    final logger = getLogger("TeamPositionNotifier");
+    logger.i("initializing a TeamPositionNotifier provider");
+
+    for (final team in teams) {
+      final dbRef = firebaseDb.child('traces').child(eventId).child(team).child('lastPosition');
+      dbRef.onValue.listen((DatabaseEvent event) {
+        final data = event.snapshot.value as String;
+        logger.i("updating $team position: ${data.toString()}");
+        final latlong = data.split(', ');
```

Comment:
you don't have to do this by yourselft. The library (LatLng if i'm not mistaken) could accept it in this form